src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -1360,6 +1371,8 @@ protected RubyMethod method(VirtualFrame frame, Object self, Object name,
                 final Object respondToMissing = respondToMissingNode
                         .call(self, "respond_to_missing?", name, dispatchConfig.ignoreVisibility);
                 if (respondToMissingProfile.profile(booleanCastNode.executeToBoolean(respondToMissing))) {
+                    /** refinements should not affect BasicObject#method_missing */
```

Comment:
It would only affect the lookup of the `method_missing` method, I guess that's the intention?
Could you clarify the comment?

Reply:
> It would only affect the lookup of the method_missing method, I guess that's the intention?

Correct; `method_missing` should not be found in activated refinements if `#respond_to_missing?` returns true (`#respond_to_missing?` is not affected by frame refinements since use `respondToMissingNode.call(..)` which is `DispatchNode.call(...)` which calls `DispatchNode.execute(frame: nil,...)`). 
So we replace callerDeclarationContext (with activated refinements) with originalDeclarationContext.

Reply:
I'd suggest:
```java
// restore the original DeclarationContext, refinements should not affect the lookup of #method_missing (MRI behavior)
```